diff --git a/node_modules/@near-js/accounts/lib/account.js b/node_modules/@near-js/accounts/lib/account.js
index c70cb3a..3af3f35 100644
--- a/node_modules/@near-js/accounts/lib/account.js
+++ b/node_modules/@near-js/accounts/lib/account.js
@@ -253,6 +253,13 @@ class Account {
      */
     functionCall({ contractId, methodName, args = {}, gas = utils_1.DEFAULT_FUNCTION_CALL_GAS, attachedDeposit, walletMeta, walletCallbackUrl, stringify, jsContract }) {
         return __awaiter(this, void 0, void 0, function* () {
+            console.log('contractId', contractId)
+            console.log('methodName', methodName)
+            console.log('args', args)
+            console.log('attachedDeposit', attachedDeposit)
+            console.log('walletMeta', walletMeta)
+            console.log('stringify', stringify)
+            console.log('jsContract', jsContract)
             this.validateArgs(args);
             let functionCallArgs;
             if (jsContract) {
@@ -263,6 +270,7 @@ class Account {
                 const stringifyArg = stringify === undefined ? transactions_1.stringifyJsonOrBytes : stringify;
                 functionCallArgs = [methodName, args, gas, attachedDeposit, stringifyArg, false];
             }
+            console.log('functionCallArgs', functionCallArgs)
             return this.signAndSendTransaction({
                 receiverId: jsContract ? this.connection.jsvmAccountId : contractId,
                 // eslint-disable-next-line prefer-spread
diff --git a/node_modules/@near-js/accounts/lib/contract.js b/node_modules/@near-js/accounts/lib/contract.js
index cda4d35..43f46a9 100644
--- a/node_modules/@near-js/accounts/lib/contract.js
+++ b/node_modules/@near-js/accounts/lib/contract.js
@@ -184,7 +184,10 @@ class Contract {
                 walletMeta: meta,
                 walletCallbackUrl: callbackUrl
             });
-            return (0, utils_1.getTransactionLastResult)(rawResult);
+            console.log('rawResult', rawResult)
+            const a = (0, utils_1.getTransactionLastResult)(rawResult)
+            console.log('(0, utils_1.getTransactionLastResult)(rawResult)', a)
+            return a;
         });
     }
 }
